src/Discovery/src/Consul/Registry/ConsulRegistration.cs

```diff
@@ -136,65 +125,6 @@ private static string[] CreateTags(ConsulDiscoveryOptions options)
         return options.Tags.ToArray();
     }
 
-    private static string GetInstanceId(ConsulDiscoveryOptions options, IApplicationInstanceInfo applicationInfo)
-    {
-        if (string.IsNullOrEmpty(options.InstanceId))
-        {
-            return NormalizeForConsul(GetDefaultInstanceId(applicationInfo));
-        }
-
-        return NormalizeForConsul(options.InstanceId);
-    }
-
-    private static string GetDefaultInstanceId(IApplicationInstanceInfo applicationInfo)
-    {
-        string appName = applicationInfo.GetApplicationNameInContext(SteeltoeComponent.Discovery, $"{ConsulDiscoveryOptions.ConfigurationPrefix}:serviceName");
-
-        string instanceId = applicationInfo.InstanceId;
-
-        if (string.IsNullOrEmpty(instanceId))
-        {
-            instanceId = Random.Shared.Next().ToString(CultureInfo.InvariantCulture);
-        }
-
-        return $"{appName}:{instanceId}";
-    }
-
-    internal static string NormalizeForConsul(string serviceId)
```

Comment:
Is this already handled in the client library?

Reply:
No, it isn't. I stepped into the library sources and found it just sends the properties over the wire, without any content checks.

![image](https://github.com/user-attachments/assets/8ba93c16-cc8f-4d69-a0a0-b8975778412b)

Reply:
So then I would _**assume**_ that's why it was here... does the Consul server not actually care about these characters being used (anymore)?

Reply:
Perhaps, but it's probably there for a reason. Even if the latest version of Consul is more relaxed, we'd be breaking against old versions for no good reason.

Reply:
Never mind, I missed where this validation moved to, there's no problem here